lib/puppet/ssl/state_machine.rb

```diff
@@ -211,10 +214,11 @@ def next_state
   #
   class Done < SSLState; end
 
-  attr_reader :waitforcert
+  attr_reader :waitforcert, :wait_deadline
 
-  def initialize(waitforcert: Puppet[:waitforcert])
+  def initialize(waitforcert: Puppet[:waitforcert], maxwaitforcert: Puppet[:maxwaitforcert])
     @waitforcert = waitforcert
+    @wait_deadline = Time.now.to_i + maxwaitforcert
   end
```

Comment:
Just a curiosity - what does an unlimited value in maxwaitforcert do here?  Is the result of the + operation just the max integer?